civiform/civiform
# Add "between" operator for eligibility and visibility predicates

### Description

Add a "between" operator for eligibility and visibility predicates.

* The new operator supports date, number, and currency questions
* Enhance the UI to support two input boxes for this operator
* Update the existing "age is between" operator to use the new UI (backwards-compatibly), and remove the help text for the old format
* Refactor the formatPredicateValue method from ProgramPredicateConfigureView into the FormattedPredicateValue class so it's easier to unit test, and add unit tests.
* Add/update tests to cover relevant new cases. Several browser screenshots change due to additional questions in the predicate tests

Design doc: https://docs.google.com/document/d/1KoBkiPlq-AjnPEtnUQ9RufgGYzTDzRq-ki1QAmgOnz0/edit

## Release notes

Add a "between" operator for eligibility and visibility conditions on date, number, and currency questions.

### Checklist

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)


#### User visible changes


### Instructions for manual testing

* Create visibility and/or eligibility conditions with the "is between" operator for questions of type date, currency, and number
* Edit a visibility or eligibility condition created on an earlier version of Civiform with a date question and the "age is between" operator. The values, previously entered in one text input as e.g. "18,30", should now show up in the two text boxes separately. The condition should render the same either way (e.g. "Screen 1 is eligible if "Applicant Date of Birth" date age is between 18 and 30").

### Issue(s) this completes

Fixes #6109
